packages/puppeteer-core/src/common/bidi/Page.ts

```diff
@@ -199,6 +202,64 @@ export class Page extends PageBase {
       return retVal;
     });
   }
+
+  override async pdf(options: PDFOptions = {}): Promise<Buffer> {
+    const {path = undefined} = options;
+    const params = this._getPDFOptions(options);
+    const {result} = await waitWithTimeout(
+      this.#context.connection.send('browsingContext.print', {
+        context: this.#context._contextId,
+        background: params.printBackground,
+        margin: params.margin,
+        orientation: params.landscape ? 'landscape' : 'portrait',
+        page: {
+          width: params.width,
+          height: params.height,
+        },
+        pageRanges: params.pageRanges.split(', '),
+        scale: params.scale,
+        shrinkToFit: !params.preferCSSPageSize,
```

Comment:
@Lightning00Blade do you recall why we decided to default shrinkToFit to preferCSSPageSize?

Reply:
Sadly I don't recall